hangman/lib/game/game.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Hangman.Game do
+
+#Public Functions
+
+    def new_game() do
+        initialize_game( retrieve_word() )        
+    end
+
+    def new_game(word) do
+        initialize_game( word |> String.split("", trim: true) )
+    end
+
+    def tally( game ) do
+        retrieve_tally( game )
+    end
+
+    def make_move( game, guess ) do
+        process_guess( game, guess )
+    end
+
+#Private Functions
+
+    defp initialize_game( word ) do
+        %GameState{ letters:      Enum.map( word, fn _ -> "_" end), 
+                    hangman_word: word }
+    end
+
+    defp retrieve_tally( %GameState{} = game ) do
+        %{ game_state: game.game_state,
+           turns_left: game.turns_left,
+           letters:    game.letters,
+           used:       game.used,
+           last_guess: game.last_guess }    
+    end
+
+    defp retrieve_word() do
+        Dictionary.WordList.random_word() |> String.split( "", trim: true ) 
+    end
+
+    defp process_guess( game, guess ) do
+        game = game |> guess( Enum.member?(game.hangman_word, guess), 
+                              Enum.member?(game.used, guess), 
+                              guess )
+
+        game = game |> game_over?( Enum.member?(game.letters, "_") )
+
```

Comment:
## -2 idiomatic

Couldn't these have been a single pipeline:

~~~ elixir
game = game
       |> guess( Enum.member?(game.hangman_word, guess),  Enum.member?(game.used, guess),  guess )
       |> game_over?( Enum.member?(game.letters, "_") )
~~~

You could even use a cool Elixir feature:

~~~ elixir
game = 
    with in_word = Enum.member?(game.hangman_word, guess),
         used = Enum.member?(game.used, guess),  guess ) 
    do: game |> guess( in_word, used) |> game_over?( Enum.member?(game.letters, "_") )
~~~